src/main/java/at/fhhagenberg/esd/sqe/ws20/gui/ECCController.java

```diff
@@ -24,248 +20,314 @@
+    @FXML
+    private Label lNextTarFloor;
+    @SuppressWarnings("unused")
+    @FXML
+    private Label lDirection;
+    @SuppressWarnings("unused")
+    @FXML
+    private Label lSpeed;
+    @SuppressWarnings("unused")
+    @FXML
+    private Label lWeight;
+    @SuppressWarnings("unused")
+    @FXML
+    private ImageView ivDoorStateClosed;
+    @SuppressWarnings("unused")
+    @FXML
+    private ImageView ivDoorStateOpen;
+    @SuppressWarnings("unused")
+    @FXML
+    private ToggleButton tbtnOperationMode;
+    @SuppressWarnings("unused")
+    @FXML
+    private Label lTargetFloor;
+    @SuppressWarnings("unused")
+    @FXML
+    private ComboBox<String> cbTargetFloor;
+    @SuppressWarnings("unused")
+    @FXML
+    private Button btnGo;
+    @SuppressWarnings("unused")
+    @FXML
+    private TextArea taErrorLog;
+    @SuppressWarnings("unused")
+    @FXML
+    private GridPane gElevator;
+    @SuppressWarnings("unused")
+    @FXML
+    private GridPane gElevatorFloors;
+    @SuppressWarnings("unused")
+    @FXML
+    private Label lTopFloor;
+    @SuppressWarnings("unused")
+    @FXML
+    private Label lGroundFloor;
+    @SuppressWarnings("unused")
+    @FXML
+    private ImageView ivGElvDirUp;
+    @SuppressWarnings("unused")
+    @FXML
+    private ImageView ivGElvDirDown;
+    @SuppressWarnings("unused")
+    @FXML
+    private Label lElvCurFloor;
+    @SuppressWarnings("unused")
+    @FXML
+    private Rectangle recElevator;
+    @SuppressWarnings("unused")
+    @FXML
+    private Group groupElevator;
+    private FloorState[] floors;
+    private ReadOnlyIntegerProperty currentElevator;
+    private ReadOnlyIntegerProperty selectedFloor;
+    private IElevatorWrapper model;
+    private GeneralInformation info;
+
+    private static ImageView createFloorImageView(String path, ObservableValue<Boolean> visible) {
+        File file = new File(path);
+        Image image = new Image(file.toURI().toString());
+        ImageView iv = new ImageView();
+        iv.setPreserveRatio(true);
+        iv.setFitHeight(20);
+        iv.setImage(image);
+        iv.visibleProperty().bind(visible);
+
+        return iv;
+    }
+
+    private void log(String message)
+    {
+        errorText.set(errorText.get() + message + "\n");
+    }
+
+    private void log(Throwable e, int level)
+    {
+        final String indent = "  ".repeat(level);
+        log(indent + e.getMessage().replace("\n", "\n" + indent));
+        if (e.getCause() != null)
+            log(e.getCause(), level+1);
+    }
+
+    private void log(Throwable e)
+    {
+        log(e, 0);
+    }
+
+    public void setModel(IElevatorWrapper model) {
+        if (model == null)
+        {
+            return;
+        }
+        this.model = model;
+
+        //Init elevator floors
+        try {
+            info = model.queryGeneralInformation();
+        } catch (Exception e) {
+            log(new RuntimeException("Failed to query model!", e));
+            this.model = null;
```

Comment:
The model implementation of #7 reconnects automatically when calling its methods, so it would not be needed the remove it.
Either way, where would you set a new model?
When merged with #7 you can distinguish between the two provided custom exception types for a different behaviour when the model is disconnected.

Reply:
> The model implementation of #7 reconnects automatically when calling its methods, so it would not be needed the remove it.

is there a way to detect if a model has been reconnected? I need to setup the floors and elevator in `setModel`. This won't work if the communication with the system doesn't work.

> When merged with #7 you can distinguish between the two provided custom exception types for a different behaviour when the model is disconnected.

I'll add functionality to disable the ui while the connection is broken

Reply:
> is there a way to detect if a model has been reconnected? I need to setup the floors and elevator in setModel. This won't  work if the communication with the system doesn't work.
Right now the reconnecting functionality is completely transparent and therefore not queryable. Basically, every call to the model will try to reconnect if needed (and throw an exception if it fails), and each failed call to the model will disconnect itself. So, it should be pretty easy to track anyway.